var/spack/repos/builtin/packages/camp/package.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Camp(CMakePackage, CudaPackage):
+    """
+    Compiler agnostic metaprogramming library providing concepts,
+    type operations and tuples for C++ and cuda
+    """
+
+    homepage = "https://github.com/LLNL/camp"
+    git      = "https://github.com/LLNL/camp.git"
+
+    version('master', branch='master', submodules='True')
+    version('0.1.0', tag='v0.1.0', submodules='True')
+
+    depends_on('cmake@3.8:', type='build')
+    depends_on('cmake@3.9:', type='build', when="+cuda")
+
+    def cmake_args(self):
+        spec = self.spec
+
+        options = []
+
+        if '+cuda' in spec:
+            options.extend([
+                '-DENABLE_CUDA=ON',
+                '-DCUDA_TOOLKIT_ROOT_DIR=%s' % (spec['cuda'].prefix)])
+
+            if not spec.satisfies('cuda_arch=none'):
+                cuda_arch = spec.variants['cuda_arch'].value
+                options.append('-DCUDA_ARCH=sm_{0}'.format(cuda_arch[0]))
+                flag = '-arch sm_{0}'.format(cuda_arch[0])
+                options.append('-DCMAKE_CUDA_FLAGS:STRING={0}'.format(flag))
+        else:
+            options.append('-DENABLE_CUDA=OFF')
+
+        options.append('-DENABLE_TESTS={0}'.format(
+            'ON' if self.run_tests else 'OFF'))
```

Comment:
If I understand correctly, setting this CMake option results in building additional files for this package (if I'm wrong about that, no need to keep reading). `self.run_tests` is controlled by the `--test`/`--run-tests` options of `spack install` (see `spack install --help`).  This doesn't actually change the Spec (i.e. the DAG hash remains the same). Generally if any choice changes the contents of the install prefix, we want to record that in the Spec.

My recommendation is to always set `-DENABLE_TESTS=ON` (don't condition it on `run_tests`). However, if the tests are expensive to build, some packages add a variant to control whether the tests are built. in that case, the variant value changes the hash depending on whether the package is built with tests or not.

Other notes:

* Dependencies recorded with `depends_on` can set `test` as the deptype. Test dependencies do not change the hash of the root spec (the distinction here is that building the test dependency isn't expected to change the contents of the root prefix).
* The wording of some of these options is changing in https://github.com/spack/spack/pull/15702
* I see at least one package (`axom`) in the builtin repo which follows this behavior, which I think would be worth changing (not in your PR that is, my point being that not every package is an example of the best way to write a new package).

Reply:
Thanks for the information. Yes, this was also leftover code borrowed from the chai package file (you noted that axom is similar). 

In any case, I changed it to always be ON.